### PR TITLE
 Team label is no longer required; fixes #550

### LINKED ISSUE #550: Fjerne krav om team-label ved deploy
Team kan 100% sikkert utledes fra hvilket namespace appen deployes inn i.

`.labels.team` kan fjernes fra deploy-krav, men bør antakelig settes igjen av Naiserator i tilfelle vi har interne eller eksterne systemer som leser denne verdien.

### PR DESCRIPTION
**Pull request by kimtore**
* `.metadata.labels.team` is no longer required for Application and Naisjob
* Remove team labels from all golden file tests
* Set the team label on Application and Naisjob when resource is persisted back to cluster
* Team label is also copied to sub-resources as before

### CODE REVIEW

**Comment by jhrv** [comment]
🎉🏆

File: pkg/resourcecreator/horizontalpodautoscaler/horizontalpodautoscaler_test.go [hunk 1]
```diff
@@ -134,7 +134,7 @@ func TestHorizontalPodAutoscaler(t *testing.T) {
 
 	t.Run("should add both cpu and kafka metric when both are set", func(t *testing.T) {
 		app := fixtures.MinimalApplication()
-		topic := fmt.Sprintf("%s.mytopic", fixtures.ApplicationTeam)
+		topic := fmt.Sprintf("%s.mytopic", fixtures.ApplicationNamespace)
```
Comment (Reasonable-Solutions) [inline pkg/resourcecreator/horizontalpodautoscaler/horizontalpodautoscaler_test.go:137]: 👏

File: pkg/scheme/listers.go [hunk 2]
```diff
@@ -31,6 +32,7 @@ func GenericListers() []client.ObjectList {
 		&corev1.SecretList{},
 		&corev1.ServiceAccountList{},
 		&corev1.ServiceList{},
+		&policyv1.PodDisruptionBudgetList{},
```
Comment (Reasonable-Solutions) [inline pkg/scheme/listers.go:35]: suddenly a new lister appears? How come?

**Review by Reasonable-Solutions — APPROVED** [review]

**Review by Reasonable-Solutions — APPROVED** [review]

File: pkg/scheme/listers.go (hunk 2 above)
Comment (kimtore) [reply pkg/scheme/listers.go:35]: Got warnings in one of the test suites (which should supposedly be an error, in previous versions)

File: pkg/resourcecreator/horizontalpodautoscaler/horizontalpodautoscaler_test.go (hunk 1 above)
Comment (kimtore) [reply pkg/resourcecreator/horizontalpodautoscaler/horizontalpodautoscaler_test.go:137]: Yes, well, now the two are the same, so it doesn't really matter.
But it does look dangerous :p

**Pull request merged**